File: lsd-interceptors/src/main/java/io/lsdconsulting/interceptors/http/naming/AlwaysAppName.java
package io.lsdconsulting.interceptors.http.naming;

import io.lsdconsulting.interceptors.common.AppName;
import lombok.RequiredArgsConstructor;
import lombok.Value;

@Value
@RequiredArgsConstructor
public class AlwaysAppName implements SourceNameMappings {
    AppName appName;
    
    @Override
    public String mapForPath(String path) {
        return appName.getValue();
    }
}

File: lsd-interceptors/src/main/kotlin/io/lsdconsulting/interceptors/http/naming/AlwaysAppName.kt
package io.lsdconsulting.interceptors.http.naming

import io.lsdconsulting.interceptors.common.AppName

data class AlwaysAppName(
    var appName: AppName
) : SourceNameMappings {

    override fun mapForPath(path: String): String {
        return appName.value
    }
}
